base_layer/core/src/base_node/sync/header_sync/synchronizer.rs

```diff
@@ -459,9 +464,9 @@ impl<'a, B: BlockchainBackend + 'static> HeaderSynchronizer<'a, B> {
                 );
                 continue;
             }
-            self.header_validator.validate(&header)?;
-            let current_height = header.height;
-            self.db.write_transaction().insert_header(header).commit().await?;
+            let chain_header = self.header_validator.validate_and_calculate_metadata(header)?;
```

Comment:
There's a bit of a name mismatch here. I have been toying with the idea of renaming `*AccumulatedData` to `*Metadata`. Thoughts?

Reply:
I think `Metadata` is a little clearer